elements/rh-dialog/rh-dialog.ts

```diff
@@ -80,93 +84,107 @@ export class RhDialog extends LitElement {
-  @query('#close-button') private closeButton?: HTMLElement | null;
+  @query('#dialog') private dialog!: HTMLDialogElement;
+  @query('#content') private content!: HTMLElement;
+  @query('#close-button') private closeButton!: HTMLElement;
 
   #headerId = getRandomId();
   #triggerElement: HTMLElement | null = null;
   #header: HTMLElement | null = null;
   #body: Element[] = [];
   #headings: Element[] = [];
   #cancelling = false;
+  #lastTabbable: HTMLElement = this.closeButton;
 
   #slots = new SlotController(this, null, 'header', 'description', 'footer');
 
   connectedCallback() {
     super.connectedCallback();
-    this.addEventListener('keydown', this.onKeydown);
+    this.addEventListener('keydown', this.#onKeyDown);
     this.addEventListener('click', this.onClick);
   }
 
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('keydown', this.#onKeyDown);
+    this.#triggerElement?.removeEventListener('click', this.onTriggerClick);
+  }
+
   render() {
     const headerId = (this.#header || this.#headings.length) ? this.#headerId : undefined;
-    const headerLabel = this.#triggerElement ? this.#triggerElement.innerText : undefined;
+    const triggerLabel = this.#triggerElement ? this.#triggerElement.innerText : undefined;
     const hasHeader = this.#slots.hasSlotted('header');
     const hasDescription = this.#slots.hasSlotted('description');
     const hasFooter = this.#slots.hasSlotted('footer');
-
     const { mobile } = this.#screenSize;
     return html`
-      <rh-surface id="rhds-wrapper"
-                           class="${classMap({ mobile })}"
-                           color-palette="lightest">
-        <section ?hidden=${!this.open}>
-          <div id="overlay" part="overlay" ?hidden=${!this.open}></div>
-          <div id="dialog"
-               part="dialog"
-               tabindex="0"
-               role="dialog"
-               aria-labelledby=${ifDefined(headerId)}
-               aria-label=${ifDefined(headerLabel)}
-               ?hidden="${!this.open}">
-            <div id="container">
-              <div id="content" part="content" class=${classMap({ hasHeader, hasDescription, hasFooter })}>
-                <header part="header">
-                  <slot name="header"></slot>
-                  <div part="description" ?hidden=${!hasDescription}>
-                    <slot name="description"></slot>
-                  </div>
-                </header>
+      <div id="rhds-wrapper" class="${classMap({ mobile })}">
+        <!-- @deprecated: 👇 use public vars for the backdrop pseudo-element instead. -->
+        <div id="overlay"
+             part="overlay"
+             ?hidden=${!this.open}>
+        </div>
+        <rh-surface class=${classMap({ hasHeader, hasDescription, hasFooter })}
+                    ?hidden=${!this.open}
+                    color-palette="${ifDefined(this.type === 'video' ? undefined : 'lightest')}">
```

Comment:
```Suggestion
                    color-palette="lightest'"
```
As discussed in Slack, `undefined` for color-palette defaults to lightest.

Reply:
Looks like this attribute is required for dark color palettes.

If you open the demo locally (after removing line 152) and switch to a dark{er/est} color palette without the attribute, the close icon is invisible when focused and light gray when not focused.